srtcore/srt.h

```diff
@@ -185,6 +192,29 @@ typedef enum SRT_SOCKOPT {
    SRTO_PACKETFILTER = 60          // Add and configure a packet filter
 } SRT_SOCKOPT;
 
+#ifdef __cplusplus
+
+    // In C++, declare the value of THE SAME enum type, but as a static constant
+    #define SRT_DECLARE_DEPRECATED_OPT(name, value) \
+        const SRT_SOCKOPT name SRT_ATR_DEPRECATED = (SRT_SOCKOPT)value;
+
+#elif defined(__GNUC__) && (__GNUC__ >= 6)
+
+    // deprecated enums are supported only since gcc 6.
+    // When not C++ and not gcc 6, simply define nothing
+
+    // In C, declare them as a separate enum, as it's not an error in C to mix enum types
+    typedef enum SRT_SOCKOPT_DEPRECATED
+    {
+    #define SRT_DECLARE_DEPRECATED_OPT(name, value) \
+        name SRT_ATR_DEPRECATED = value,
+
+#else
+#define SRT_NO_DEPRECATED_ENUM
+#define SRT_DECLARE_DEPRECATED_OPT(name, value) \
```

Comment:
This does not look right: no deprecated values will be defined. Also I think we still want to have `enum SRT_SOCKOPT_DEPRECATED {}` for C compatibility.

Reply:
Ok, I've rewritten it completely. Tested with an old compiler in C, C++ and C++11 modes. Reports the misused deprecated option perfectly in all cases.

Reply:
With gcc 4.4.1 deprecated option is not reported. Instead, I get  
> warning: 'SRT_SOCKOPT_DEPRECATED' is deprecated

Reply:
Intended is that you get a warning when this option is **used**, no matter what the warning says. If it works this way, that's good enough. I checked 4.4.1 and it reports exactly the line where the deprecated enum is used - even though the message isn't too readable.